core/server/worker/src/main/java/alluxio/worker/dora/PagedFileReader.java

```diff
@@ -77,7 +77,7 @@ public ByteBuffer read(long offset, long length) throws IOException {
 
     // cap length to the remaining of block, as the caller may pass in a longer length than what
     // is left in the block, but expect as many bytes as there is
-    length = Math.min(length, mPos - offset);
+//    length = Math.min(length, mPos - offset);
```

Comment:
@dbw9580 I wonder why we have this limit? for the new data reader would this always be 0?

Reply:
i dont think u can comment this out, read workflow will still issue read chunk by chunk at dora worker end, it can not directly be the length called on this read() func, i think in the load func, u will have to do the same in PacketReader, to load chunk by chunk (or actually page by page as unlike read, load behavior is totally controlled by us )

Reply:
Yes, this is actually a quirky part of the interface `BlockReader`. See the comments on `alluxio.worker.block.io.BlockReader#read` I left a few whiles back. Instead of returning -1 for EOF, it returns a empty ByteBuf as a signal to `alluxio.worker.grpc.BlockReadHandler.DataReader`. 

The suggestion is to use the `transferTo` method instead.